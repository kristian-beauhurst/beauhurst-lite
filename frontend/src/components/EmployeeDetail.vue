<template>
  <div class="employee-detail">
    <div v-if="loading" class="loading">
      Loading...
    </div>
    <div v-else-if="error" class="error">
      {{ error }}
    </div>
    <div v-else-if="employee" class="content">
      <div class="header">
        <h1>{{ employee.name }}</h1>
        <div class="meta">
          <span class="job-title">{{ employee.job_title }}</span>
          <span class="company" v-if="employee.company">
            <router-link :to="'/companies/' + employee.company.id">
              {{ employee.company.name }}
            </router-link>
          </span>
        </div>
      </div>

      <div class="contact">
        <h2>Contact Information</h2>
        <div class="contact-grid">
          <div class="contact-item">
            <i class="fas fa-envelope"></i>
            <a :href="'mailto:' + employee.email">{{ employee.email }}</a>
          </div>
          <div class="contact-item" v-if="employee.phone_number">
            <i class="fas fa-phone"></i>
            <a :href="'tel:' + employee.phone_number">{{ employee.phone_number }}</a>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8000/api/v1';

export default {
  name: 'EmployeeDetail',
  props: {
    id: {
      type: [String, Number],
      required: true
    }
  },
  data() {
    return {
      employee: null,
      loading: true,
      error: null
    };
  },
  async created() {
    try {
      const response = await axios.get(`${API_BASE_URL}/employees/${this.id}/`);
      this.employee = response.data;
    } catch (error) {
      this.error = 'Failed to load employee details';
      console.error('Error loading employee:', error);
    } finally {
      this.loading = false;
    }
  }
};
</script>

<style scoped>
.employee-detail {
  max-width: 1200px;
  margin: 0 auto;
  padding: 2rem;
}

.loading, .error {
  text-align: center;
  padding: 2rem;
  font-size: 1.2rem;
}

.error {
  color: #dc3545;
}

.header {
  margin-bottom: 2rem;
}

.header h1 {
  font-size: 2.5rem;
  color: #2c3e50;
  margin-bottom: 1rem;
}

.meta {
  display: flex;
  gap: 1rem;
  color: #666;
}

.job-title {
  font-size: 1.2rem;
  color: #2c3e50;
}

.company a {
  color: #007bff;
  text-decoration: none;
}

.contact {
  margin-top: 3rem;
}

.contact h2 {
  color: #2c3e50;
  margin-bottom: 1rem;
}

.contact-grid {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
  gap: 1.5rem;
}

.contact-item {
  display: flex;
  align-items: center;
  gap: 0.5rem;
  padding: 1rem;
  border: 1px solid #ddd;
  border-radius: 8px;
  background-color: white;
}

.contact-item i {
  color: #666;
  font-size: 1.2rem;
}

.contact-item a {
  color: #007bff;
  text-decoration: none;
}
</style> 