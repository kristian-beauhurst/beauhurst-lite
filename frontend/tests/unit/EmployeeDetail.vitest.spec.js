import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'
import EmployeeDetail from '@/components/EmployeeDetail.vue'
import axios from 'axios'
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest'

vi.mock('axios')

const router = createRouter({
  history: createWebHistory(),
  routes: [
    { path: '/employees/:id', component: EmployeeDetail }
  ]
})

const mockEmployeeData = {
  id: 1,
  name: 'John Smith',
  job_title: 'Senior Software Engineer',
  email: 'john.smith@acme.com',
  phone_number: '+44 123 456 7890',
  gender: 'M',
  company: {
    id: 1,
    name: 'Acme Corporation'
  }
}

const mockEmployeeWithoutPhone = {
  id: 2,
  name: 'Jane Doe',
  job_title: 'Product Manager',
  email: 'jane.doe@acme.com',
  phone_number: null,
  gender: 'F',
  company: {
    id: 1,
    name: 'Acme Corporation'
  }
}

describe('EmployeeDetail.vue (Vitest)', () => {
  let wrapper

  beforeEach(() => {
    vi.clearAllMocks()
  })

  afterEach(() => {
    if (wrapper) {
      wrapper.unmount()
    }
  })

  it('renders loading state correctly', async () => {
    axios.get.mockImplementation(() => new Promise(() => {}))

    wrapper = mount(EmployeeDetail, {
      props: { id: 1 },
      global: {
        plugins: [router]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders error state correctly', async () => {
    axios.get.mockRejectedValueOnce(new Error('Network error'))

    wrapper = mount(EmployeeDetail, {
      props: { id: 1 },
      global: {
        plugins: [router]
      }
    })

    await wrapper.vm.$nextTick()
    await new Promise(resolve => setTimeout(resolve, 0))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders employee data correctly', async () => {
    axios.get.mockResolvedValueOnce({ data: mockEmployeeData })

    wrapper = mount(EmployeeDetail, {
      props: { id: 1 },
      global: {
        plugins: [router]
      }
    })

    await wrapper.vm.$nextTick()
    await new Promise(resolve => setTimeout(resolve, 0))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders employee without phone number correctly', async () => {
    axios.get.mockResolvedValueOnce({ data: mockEmployeeWithoutPhone })

    wrapper = mount(EmployeeDetail, {
      props: { id: 2 },
      global: {
        plugins: [router]
      }
    })

    await wrapper.vm.$nextTick()
    await new Promise(resolve => setTimeout(resolve, 0))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders employee without company correctly', async () => {
    const employeeWithoutCompany = {
      ...mockEmployeeData,
      company: null
    }
    axios.get.mockResolvedValueOnce({ data: employeeWithoutCompany })

    wrapper = mount(EmployeeDetail, {
      props: { id: 3 },
      global: {
        plugins: [router]
      }
    })

    await wrapper.vm.$nextTick()
    await new Promise(resolve => setTimeout(resolve, 0))

    expect(wrapper.html()).toMatchSnapshot()
  })
}) 