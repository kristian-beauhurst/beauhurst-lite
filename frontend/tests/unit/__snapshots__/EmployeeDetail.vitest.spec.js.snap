// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmployeeDetail.vue (Vitest) > renders employee data correctly 1`] = `
"<div data-v-f333d322="" class="employee-detail">
  <div data-v-f333d322="" class="content">
    <div data-v-f333d322="" class="header">
      <h1 data-v-f333d322="">John Smith</h1>
      <div data-v-f333d322="" class="meta"><span data-v-f333d322="" class="job-title">Senior Software Engineer</span><span data-v-f333d322="" class="company"><a data-v-f333d322="" href="/companies/1" class="">Acme Corporation</a></span></div>
    </div>
    <div data-v-f333d322="" class="contact">
      <h2 data-v-f333d322="">Contact Information</h2>
      <div data-v-f333d322="" class="contact-grid">
        <div data-v-f333d322="" class="contact-item"><i data-v-f333d322="" class="fas fa-envelope"></i><a data-v-f333d322="" href="mailto:john.smith@acme.com">john.smith@acme.com</a></div>
        <div data-v-f333d322="" class="contact-item"><i data-v-f333d322="" class="fas fa-phone"></i><a data-v-f333d322="" href="tel:+44 123 456 7890">+44 123 456 7890</a></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`EmployeeDetail.vue (Vitest) > renders employee without company correctly 1`] = `
"<div data-v-f333d322="" class="employee-detail">
  <div data-v-f333d322="" class="content">
    <div data-v-f333d322="" class="header">
      <h1 data-v-f333d322="">John Smith</h1>
      <div data-v-f333d322="" class="meta"><span data-v-f333d322="" class="job-title">Senior Software Engineer</span>
        <!--v-if-->
      </div>
    </div>
    <div data-v-f333d322="" class="contact">
      <h2 data-v-f333d322="">Contact Information</h2>
      <div data-v-f333d322="" class="contact-grid">
        <div data-v-f333d322="" class="contact-item"><i data-v-f333d322="" class="fas fa-envelope"></i><a data-v-f333d322="" href="mailto:john.smith@acme.com">john.smith@acme.com</a></div>
        <div data-v-f333d322="" class="contact-item"><i data-v-f333d322="" class="fas fa-phone"></i><a data-v-f333d322="" href="tel:+44 123 456 7890">+44 123 456 7890</a></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`EmployeeDetail.vue (Vitest) > renders employee without phone number correctly 1`] = `
"<div data-v-f333d322="" class="employee-detail">
  <div data-v-f333d322="" class="content">
    <div data-v-f333d322="" class="header">
      <h1 data-v-f333d322="">Jane Doe</h1>
      <div data-v-f333d322="" class="meta"><span data-v-f333d322="" class="job-title">Product Manager</span><span data-v-f333d322="" class="company"><a data-v-f333d322="" href="/companies/1" class="">Acme Corporation</a></span></div>
    </div>
    <div data-v-f333d322="" class="contact">
      <h2 data-v-f333d322="">Contact Information</h2>
      <div data-v-f333d322="" class="contact-grid">
        <div data-v-f333d322="" class="contact-item"><i data-v-f333d322="" class="fas fa-envelope"></i><a data-v-f333d322="" href="mailto:jane.doe@acme.com">jane.doe@acme.com</a></div>
        <!--v-if-->
      </div>
    </div>
  </div>
</div>"
`;

exports[`EmployeeDetail.vue (Vitest) > renders error state correctly 1`] = `
"<div data-v-f333d322="" class="employee-detail">
  <div data-v-f333d322="" class="error">Failed to load employee details</div>
</div>"
`;

exports[`EmployeeDetail.vue (Vitest) > renders loading state correctly 1`] = `
"<div data-v-f333d322="" class="employee-detail">
  <div data-v-f333d322="" class="loading"> Loading... </div>
</div>"
`;
